sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala

```diff
@@ -52,6 +52,23 @@ class DataTypeSuite extends SparkFunSuite {
     assert(StructField("b", LongType, false) === struct("b"))
   }
 
+  test("construct with add from StructField with comments") {
```

Comment:
Since this PR also add two function calls `add` in `StructType`. This test case also covers the interface changes. 

I am fine if you think that test is useless. Let me know if you are OK to keep them. Thanks!